### ISSUE TITLE
element of tuple (w, *args) inferred with 'Any' although both 'w' and 'args' have type 'str'

### ISSUE THREAD

**Issue by PatrickBourdon** [microsoft/pylance-release#991]
## Environment data
-   Windows-10-10.0.18362-SP0
-   python(3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)])
-   Visual Code Studio: 1.53.2
-   extension pylance: v2021.2.4
-   type checking mode: strict

## Expected behaviour
A problem reported.

## Actual behaviour
No problemreported.
see example below

## Code Snippet / Additional information
```python
def func1(w: str, *args: str) -> list[str]:
    y = [ z.lower() for z in (w, *args) ]
    y.append([1, 'a'])
    return y  
# pylance infers that 'z' has type 'Any', and that 'y' has type list[Any].
# Therefore lower() appears in white, and you can append anything to 'y'
# without a problem be reported.

# The issue disappears when the tuple is changed to a list:
def func2(w: str, *args: str) -> list[str]:
    return [ z.lower() for z in [w, *args] ]

```

**Comment by judej** [comment]
@erictraut, could you please look at this?

**Comment by erictraut** [comment]
It's correct either way. Currently, a tuple expression that includes an unpacked open-ended (unknown length) tuple is translated to a `tuple[Any, ...]`. I can add additional logic to combine the types in the particular case where all of the other entries in the tuple share the same type as the unpacked open-ended tuple. It's a very special case, but it's reasonable to expect that they will be combined correctly. The list expression logic contains logic to handle this special case already.

**Comment by erictraut** [comment]
This will be fixed in the next release. Here's the commit: https://github.com/microsoft/pyright/commit/b65928c00ec50eafa2203ca387077bab45404ccf.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.3.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202130-3-march-2021
